Windows: The same container cannot be recreated in the same Pod

<!--
If you are reporting a new issue, make sure that we do not have any duplicates
already open. You can ensure this by searching the issue list for this
repository. If there is a duplicate, please close your issue and add a comment
to the existing issue instead.
-->

**Description**

If a container from a Pod is deleted, it can no longer be created with the same configuration on Windows due to this error:

```
time="2021-02-26T14:36:42Z" level=fatal msg="creating container: rpc error: code = NotFound desc = failed to get sandbox container task: no running task found: task fd321b0a09e7d3509f4be47b8c043b174b8f16d2069b39f2837952ba1c17074c not found: not found"
```

This error can be observed while running the CRI integration test ``TestContainerRestart``. The steps executed by the test are the same as the ones below.

<!--
Briefly describe the problem you are having in a few paragraphs.
-->

**Steps to reproduce the issue:**

```
crictl --runtime-endpoint=npipe://./pipe/containerd-containerd pull docker.io/claudiubelu/pause:3.4.1
$POD_ID=(crictl --runtime-endpoint=npipe://./pipe/containerd-containerd runp .\pod.json)
$CONTAINER_ID=(crictl --runtime-endpoint=npipe://./pipe/containerd-containerd create $POD_ID .\container1.json .\pod.json)
crictl --runtime-endpoint=npipe://./pipe/containerd-containerd start $CONTAINER_ID
crictl --runtime-endpoint=npipe://./pipe/containerd-containerd stop $CONTAINER_ID
crictl --runtime-endpoint=npipe://./pipe/containerd-containerd rm $CONTAINER_ID

# the error occurs here on master:
$CONTAINER_ID=(crictl --runtime-endpoint=npipe://./pipe/containerd-containerd create $POD_ID .\container1.json .\pod.json)
```

pod.json:

```
{
  "metadata": {
    "name": "sandbox",
    "namespace": "default",
    "attempt": 1,
    "uid": "hdishd83djaidwnduwk28bcsb"
  },
  "logDirectory": "/tmp"
}

```

container1.json:

```
{
  "metadata": {
      "name": "busybox"
  },
  "image":{
      "image": "docker.io/claudiubelu/pause:3.4.1"
  },
  "command": [],
  "args": []
}
```


**Describe the results you received:**

The last step will generate the following error:

```
time="2021-02-26T14:36:42Z" level=fatal msg="creating container: rpc error: code = NotFound desc = failed to get sandbox container task: no running task found: task fd321b0a09e7d3509f4be47b8c043b174b8f16d2069b39f2837952ba1c17074c not found: not found"
```

It is noteworthy that this error occurs with a ``containerd`` built from master. This error does not occur in version ``v1.4.3``.


**Describe the results you expected:**

The opperation should have succeeded.


**Output of `containerd --version`:**

```
containerd github.com/containerd/containerd v1.5.0-beta.1-4-ge288feacf.m e288feacf35b202d483665fbd296113897427aca.m
```

This scenario works with the following containerd version:

```
containerd github.com/containerd/containerd v1.4.3 269548fa27e0089a8b8278fc4fc781d7f65a939b

```

**Any other relevant information:**

Some more updates: This affects the newly introduced cri-integration tests, specifically the "TestContainerRestart" test. The test creates a pod & a container in that pod. When it tries to restart the container it will error out as the shim is gone & the pod in NotReady state.

`
=== RUN   TestContainerRestart
    container_restart_test.go:28: Create a pod config and run sandbox container
    common.go:115: Image "k8s.gcr.io/pause:3.5" already exists, not pulling.
    container_restart_test.go:39: Create a container config and run container in a pod
    container_restart_test.go:56: Restart the container with same config
E0525 11:42:20.019650    3056 remote_runtime.go:241] CreateContainer in sandbox "9ae354013a377b17848b6dccf7f8e7975d8769b6b125e6de631d85679b4adad5" from runtime service failed: rpc error: code = NotFound desc = failed to get sandbox container task: no running task found:
 task 9ae354013a377b17848b6dccf7f8e7975d8769b6b125e6de631d85679b4adad5 not found: not found
    container_restart_test.go:61:
                Error Trace:    container_restart_test.go:61
                Error:          Received unexpected error:
                                rpc error: code = NotFound desc = failed to get sandbox container task: no running task found: task 9ae354013a377b17848b6dccf7f8e7975d8769b6b125e6de631d85679b4adad5 not found: not found
                Test:           TestContainerRestart
E0525 11:42:20.022406    3056 remote_runtime.go:288] StopContainer "" from runtime service failed: rpc error: code = Unknown desc = an error occurred when try to find container "": Prefix can't be empty
    container_restart_test.go:53:
                Error Trace:    container_restart_test.go:53
                                                        panic.go:613
                                                        testing.go:741
                                                        container_restart_test.go:61
                Error:          Received unexpected error:
                                rpc error: code = Unknown desc = an error occurred when try to find container "": Prefix can't be empty
                Test:           TestContainerRestart
E0525 11:42:20.026415    3056 remote_runtime.go:308] RemoveContainer "" from runtime service failed: rpc error: code = Unknown desc = an error occurred when try to find container "": Prefix can't be empty
    container_restart_test.go:49:
                Error Trace:    container_restart_test.go:49
                                                        panic.go:613
                                                        testing.go:741
                                                        container_restart_test.go:61
                Error:          Received unexpected error:
                                rpc error: code = Unknown desc = an error occurred when try to find container "": Prefix can't be empty
                Test:           TestContainerRestart
--- FAIL: TestContainerRestart (8.92s)

`

--- a/go.mod
+++ b/go.mod
@@ -4,7 +4,7 @@ go 1.16
 
 require (
 	github.com/Microsoft/go-winio v0.5.0
-	github.com/Microsoft/hcsshim v0.8.17
+	github.com/Microsoft/hcsshim v0.8.18
 	github.com/containerd/aufs v1.0.0
 	github.com/containerd/btrfs v1.0.0
 	github.com/containerd/cgroups v1.0.1

